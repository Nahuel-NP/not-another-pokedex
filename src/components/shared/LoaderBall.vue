<template>
  <div class="flex flex-col justify-center items-center">
    <p class="text-xl text-white text-center font-bold mb-10">Cargando...</p>
    <img src="/pokeball.svg" alt="pokeball loader" class="max-w-32 animate-spin" />
  </div>
</template>
