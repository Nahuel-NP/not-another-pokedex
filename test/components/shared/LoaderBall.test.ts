import { mount } from "@vue/test-utils"
import LoaderBall from '@/components/shared/LoaderBall.vue';

describe('Test LoaderBall component', () => {

  test('should match spanshot', () => {

    const wrapper = mount(LoaderBall)
    expect(wrapper.html()).toMatchSnapshot();
  })

  test('img should contain animate-spin class ', () => {

    const wrapper = mount(LoaderBall)
    expect(wrapper.find('img').classes()).toContain('animate-spin')

  })
})