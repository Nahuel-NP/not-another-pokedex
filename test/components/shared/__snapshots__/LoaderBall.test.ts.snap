// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test LoaderBall component > should match spanshot 1`] = `
"<div class="flex flex-col justify-center items-center">
  <p class="text-xl text-white text-center font-bold mb-10">Cargando...</p><img src="/pokeball.svg" alt="pokeball loader" class="max-w-32 animate-spin">
</div>"
`;
